src/utils/generateStudentQRCode.js

```diff
@@ -1,11 +1,52 @@
 import qrcode from 'qrcode';
+import { createCanvas, loadImage } from 'canvas';
 
-export async function generateStudentQRCode(studentId) {
-  const qrCode = await qrcode.toDataURL(studentId, {
+export async function generateStudentQRCode(studentId, base64Image) {
+  const qrCodeOptions = {
     errorCorrectionLevel: 'H',
     width: 256,
     color: { dark: '#000000', light: '#ffffff' },
-  });
+  };
+  const qrCodeDataUrl = await qrcode.toDataURL(studentId, qrCodeOptions);
 
-  return qrCode;
+  const canvas = createCanvas(qrCodeOptions.width, qrCodeOptions.width);
+  const context = canvas.getContext('2d');
+
+  const base64ImageDimensions = {
+    width: qrCodeOptions.width * 0.3,
+    height: qrCodeOptions.width * 0.3,
+  };
+
+  const [qrCodeImageCanvasElement, base64ImageCanvasElement] = Promise.all([
+    loadImage(qrCodeDataUrl),
+    loadImage(base64Image),
+  ]);
+
+  context.drawImage(
+    qrCodeImageCanvasElement,
+    0,
+    0,
+    qrCodeOptions.width,
+    qrCodeOptions.width
+  );
+
+  const centerX = qrCodeOptions.width / 2;
+  const centerY = qrCodeOptions.width / 2;
```

Comment:
Agrupar variavéis em uma só.

```suggestion
  const centerCoords = {
    x: qrCodeOptions.width / 2,
    y: qrCodeOptions.width / 2
  };
```